package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -121,27 +135,69 @@ _proto_mbim_setup() {
 	done
 	tid=$((tid + 1))
 
-	uci_set_state network $interface tid "$tid"
+	echo "mbim[$$]" "Connected"
 
-	echo "mbim[$$]" "Connected, starting DHCP"
-	proto_init_update "$ifname" 1
-	proto_send_update "$interface"
+	if [ "$dhcp" = 0 ]; then
+		echo "mbim[$$]" "Setting up $ifname"
+		eval $(umbim $DBG -n -t $tid -d $device config | sed 's/: /=/g')
+		tid=$((tid + 1))
 
-	json_init
-	json_add_string name "${interface}_4"
-	json_add_string ifname "@$interface"
-	json_add_string proto "dhcp"
-	proto_add_dynamic_defaults
-	json_close_object
-	ubus call network add_dynamic "$(json_dump)"
-
-	json_init
-	json_add_string name "${interface}_6"
-	json_add_string ifname "@$interface"
-	json_add_string proto "dhcpv6"
-	json_add_string extendprefix 1
-	proto_add_dynamic_defaults
-	ubus call network add_dynamic "$(json_dump)"
+		proto_init_update "$ifname" 1
+		proto_send_update "$interface"
+
+		json_init
+		json_add_string name "${interface}_4"
```

Comment:
While already touching this - as I mentioned in the comment, I would selectively enable sub-interfaces basing on IP type selection.
Otherwise - LGTM.